pkg/operators/oci-handler/oci_test.go

```diff
@@ -0,0 +1,347 @@
+			expectedErrorMsg: "getting secret \"wrong-ns-secret\"",
+		},
+		{
+			name:            "large content",
+			secretName:      "largeSecret",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			secretDataKey:   corev1.DockerConfigJsonKey,
+			secretContent:   string(make([]byte, 1000)),
+			expectError:     false,
+			expectedResult:  string(make([]byte, 1000)),
+		},
+		{
+			name:            "nil data map",
+			secretName:      "nilDataSecret",
+			secretNamespace: "default",
+			secretType:      corev1.SecretTypeDockerConfigJson,
+			expectError:     false,
+			expectedResult:  "",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fcs := fake.NewSimpleClientset()
+
+			if test.secretType != "" {
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      test.secretName,
+						Namespace: test.secretNamespace,
+					},
+					Type: test.secretType,
+				}
+				if test.secretDataKey != "" {
+					secret.Data = map[string][]byte{
+						test.secretDataKey: []byte(test.secretContent),
+					}
+				}
+				_, err := fcs.CoreV1().Secrets(test.secretNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+				require.NoError(t, err, "setup: failed to create test secret")
+			}
+
+			result, err := getPullSecret(test.secretName, test.secretNamespace, fcs)
+
+			if test.expectError {
+				require.Error(t, err)
+				require.Contains(t, err.Error(), test.expectedErrorMsg)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, test.expectedResult, string(result))
+			}
+		})
+	}
+}
+
+func TestConstructTempConfig(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		wantErr    bool
+		wantConfig map[string]any
+		wantLength int
+	}{
+		{
+			name:    "Invalid no colon",
+			input:   "invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid no equal sign",
+			input:   "source:invalidannotation",
+			wantErr: true,
+		},
+		{
+			name:    "Valid datasource annotation",
+			input:   "source:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"annotations": map[string]any{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+			wantLength: 1,
+		},
+		{
+			name:    "Valid field annotation",
+			input:   "source.field:foo=bar",
+			wantErr: false,
+			wantConfig: map[string]any{
+				"datasources": map[string]any{
+					"source": map[string]any{
+						"fields": map[string]any{
+							"field": map[string]any{
+								"annotations": map[string]any{
+									"foo": "bar",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantLength: 2,
+		},
+		// TODO: Shouldn't these be invalid?
+		{
+			name:    "Invalid multiple colons",
+			input:   "source:field:foo=bar",
+			wantErr: true,
+		},
+		{
+			name:    "Invalid multiple equal signs",
+			input:   "source:foo=bar=baz",
```

Comment:
This should be valid. There might be cases where the value might be base64 encoded and therefore the value could have multiple `=` equal signs